app/src/androidTest/java/org/kiwix/kiwixmobile/main/KiwixMainRobot.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import applyWithViewHierarchyPrinting
+import org.kiwix.kiwixmobile.BaseRobot
+import org.kiwix.kiwixmobile.Findable.Text
+import org.kiwix.kiwixmobile.Findable.ViewId
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString
+
+fun kiwixMainRobo(func: KiwixMainRobot.() -> Unit) =
+  KiwixMainRobot().applyWithViewHierarchyPrinting(func)
+
+class KiwixMainRobot : BaseRobot() {
+
+  init {
+    isVisible(ViewId(R.id.navigation_container))
+  }
+
+  fun clickReaderOnBottomNav() {
+    clickOn(ViewId(R.id.readerFragment))
+  }
+
+  fun assertReaderScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.reader)))
+  }
+
+  fun clickLibraryOnBottomNav() {
+    clickOn(ViewId(R.id.libraryFragment))
+  }
+
+  fun assertLibraryScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.library)))
+  }
+
+  fun clickFileTransferIcon() {
+    clickOn(ViewId(R.id.get_zim_nearby_device))
+  }
+
+  fun assertFileTransferScreenDisplayed() {
+    isVisible(Text("Receive Files"))
+  }
+
+  fun clickDownloadOnBottomNav() {
+    clickOn(ViewId(R.id.downloadsFragment))
+  }
+
+  fun assertDownloadScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.download)))
+  }
+
+  fun clickOnGlobeIcon() {
+    clickOn(ViewId(R.id.select_language))
+  }
+
+  fun assertLanguagesScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.select_languages)))
+  }
+
+  fun clickBookmarksOnNavDrawer() {
+    clickOn(Text(getResourceString(R.string.bookmarks)))
```

Comment:
You should be able to use the TextId Findableand just pass the string ID in

Reply:
* Nope, this also didn't work. I tried with the above method, that too didn't work. I have no clue why it fails and sadly it is the only way(I think) from which we can verify the Bookmarks screen. 
* The test astonishingly fails saying: `No view found with text View Bookmarks From All Books`.
* But the test did pass initially(when I used barista, but I'm no longer in mood to use it now) 

PS: I used it `TextId` the same way it was used in ZimManageTest and HelpActivityTest

Reply:
Something weird is going on here, if this works then TextId absolutely should and also this would not be the line failing
`<string name="bookmarks">Bookmarks</string>` it would be looking for the text bookmarks not `View Bookmarks From All Books` which I assume is the `isDisplayed` assertion in the `init` of BookmarksRobot to confirm we successfully navigated there. So: Does it navigate there when you use TextID? This should be readily confirmable by observing the device as you run this test